lisp/mastodon-toot.el

```diff
@@ -118,6 +123,57 @@ Set `mastodon-toot--content-warning' to nil."
   (setq mastodon-toot--content-warning
         (not mastodon-toot--content-warning)))
 
+(defun mastodon-toot--get-mode-kbinds ()
+  "Get a list of the keybindings in the mastodon-toot-mode."
+  (remove-if-not (lambda (x) (listp x))
+		 (cadr mastodon-toot-mode-map)))
+
+(defun mastodon-toot--format-kbind (kbind)
+  "Format a single keybinding, KBIND, for display in documentation."
+  (let ((key (help-key-description (vector (car kbind)) nil))
+	(command (cdr kbind)))
+    (format "\t%s - %s" key command)))
+
+(defun mastodon-toot--format-kbinds (kbinds)
+  "Format a list keybindings, KBINDS, for display in documentation."
+  (string-join (cons "" (mapcar #'mastodon-toot--format-kbind kbinds))
+	       "\n"))
+
+(defun mastodon-toot--make-mode-docs ()
+  "Create formatted documentation text for the mastodon-toot-mode."
+  (let ((kbinds (mastodon-toot--get-mode-kbinds)))
+    (concat
+     "=================================================================\n"
+     "Compose a new toot here. The following keybindings are available:"
+     (mastodon-toot--format-kbinds kbinds)
```

Comment:
This is a clever way to do it, but we might prefer a hard coded list. I only say that because I want to show the entire keybinding (`C-c C-c` vs `C-c`) and I want a description of behavior rather than the function name.

`mastodon-toot--toggle-warning` => Add/remove content warning
`mastodon-toot--send` => Send toot
`mastodon-toot--cancel` => Cancel toot

I'm not in love with the hard coded solution. If we can do these things dynamically, go for it.